### PR TITLE
[x/programs] add rust linter

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds a github action and fixes for linting rust using clippy via the latest nightly.

### CODE REVIEW

File: scripts/rust/tests.lint.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# rustup component add rustfmt
+# rustup component add rustfmt --toolchain nightly
+# rustup component add clippy
+# rustup component add clippy --toolchain nightly
+
+rustup default stable
+cargo fmt --all --verbose -- --check
+
+rustup default nightly
+cargo +nightly fmt --all -- --config-path .rustfmt.nightly.toml --verbose --check || true
```
Comment (hexfusion) [inline scripts/rust/tests.lint.sh:22]: @samliok if we drop the `|| true` here these will run/error can you see if the output is reasonable if yes update the script? If not add a TODO to enable?

File: scripts/rust/tests.lint.sh
Comment (patrick-ogrady) [inline scripts/rust/tests.lint.sh:1]: Is there a reason this isn't in `x/programs/rust`?

**Review by samliok — DISMISSED** [review]

**Review by samliok — COMMENTED** [review]
looks good.

**Review by patrick-ogrady — APPROVED** [review]

**Review by patrick-ogrady — CHANGES_REQUESTED** [review]
Question on action

File: .github/workflows/rust-static-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+
+permissions:
```
Comment (patrick-ogrady) [inline .github/workflows/rust-static-analysis.yml:14]: Why is this required?

Comment (hexfusion) [reply .github/workflows/rust-static-analysis.yml]: removed

File: x/programs/rust/wasmlanche_sdk/src/host/mod.rs [hunk 3]
```diff
@@ -42,7 +42,12 @@ pub fn init_program_storage() -> ProgramContext {
     unsafe { ProgramContext::from(_init_program()) }
 }
 
-pub fn store_bytes(
+/// # Safety
+/// The caller must ensure that key_ptr + key_len and
+/// value_ptr + value_len point to valid memory locations.
+///
+/// Stores the bytes at value_ptr to the bytes at key ptr on the host.
```
Comment (hexfusion) [inline x/programs/rust/wasmlanche_sdk/src/host/mod.rs:49]: I believe the proper syntax is the safety notice goes below the docs for the method/fucntion

/// Docs
///
/// # Safety

ref. https://doc.rust-lang.org/src/core/ptr/const_ptr.rs.html#406

Comment (hexfusion) [reply x/programs/rust/wasmlanche_sdk/src/host/mod.rs:49]:
```suggestion
/// Stores the bytes at value_ptr to the bytes at key ptr on the host.
///
/// # Safety
/// The caller must ensure that key_ptr + key_len and
/// value_ptr + value_len point to valid memory locations.
```

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**